### PR TITLE
Bugfix Ping.cs

### PR DESCRIPTION
**Pull request by jogibear9988**
Should fix : https://bugzilla.xamarin.com/show_bug.cgi?id=42249

### CODE REVIEW

File: mcs/class/System/System.Net.NetworkInformation/Ping.cs [hunk 1]
```diff
@@ -215,7 +215,7 @@ static IPAddress GetNonLoopbackIP ()
 					return addr;
 #pragma warning restore 618
 
-			throw new InvalidOperationException ("Could not resolve non-loopback IP address for localhost");
+			return IPAddress.Loopback;
```
Comment (akoeplinger) [inline mcs/class/System/System.Net.NetworkInformation/Ping.cs]: I don't know why we need a non-loopback address, but this is not the correct way of fixing it.
You need to modify SendPrivileged() so it doesn't rely on GetNonLoopbackIP().

**Comment by jogibear9988** [comment]
When I look again on this, I think the whole Code is wrong! We need the IP wich was used for send, and not the first one with correct hostname.
But how can we find out?

**Comment by jogibear9988** [comment]
seems like netcore does not need this https://github.com/dotnet/corefx/blob/master/src/System.Net.Ping/src/System/Net/NetworkInformation/Ping.Unix.cs

**Comment by akoeplinger** [comment]
Yeah, the corefx code looks better in that regard. You can try porting the relevant pieces.

**Comment by jogibear9988** [comment]
I will try to use the corefx code. Is it not planed to migrate a this Assembly in Mono?

Which Editor can I use? I tried with VS2015, but it crashes with the Mono Solution.

**Comment by jogibear9988** [comment]
With the new change it does work. And it's the same as in corefx. They also only use the destination IP

**Comment by jogibear9988** [comment]
@akoeplinger is the patch so ok?

File: mcs/class/System/System.Net.NetworkInformation/Ping.cs (hunk 1 above)
Comment (jogibear9988) [inline mcs/class/System/System.Net.NetworkInformation/Ping.cs]: I've changed the COde

**Comment by jogibear9988** [comment]
@akoeplinger - do i still need to do changes on my patch?

**Pull request closed without merging**